src/v/debug_bundle/debug_bundle_service.cc

```diff
@@ -746,5 +783,92 @@ ss::future<> service::handle_wait_result(job_id_t job_id) {
         co_return;
     }
 }
+ss::future<> service::maybe_reload_previous_run() {
+    auto md_buf = _kvstore->get(
+      storage::kvstore::key_space::debug_bundle,
+      bytes::from_string(debug_bundle_metadata_key));
+
+    if (!md_buf) {
+        vlog(lg.debug, "No previous run detected");
+        co_return;
+    }
+
+    iobuf_parser p(std::move(*md_buf));
+    auto md = serde::read<metadata>(p);
+
+    auto run_was_successful = was_run_successful(md.get_wait_status());
+    auto& debug_bundle_file_path = md.debug_bundle_file_path;
+    auto& process_output_file_path = md.process_output_file_path;
+
+    const auto cleanup_files = [debug_bundle_file_path,
+                                process_output_file_path]() -> ss::future<> {
+        if (co_await ss::file_exists(debug_bundle_file_path)) {
+            co_await ss::remove_file(debug_bundle_file_path);
+        }
+
+        if (co_await ss::file_exists(process_output_file_path)) {
+            co_await ss::remove_file(process_output_file_path);
+        }
+    };
+
+    if (
+      run_was_successful && !co_await ss::file_exists(debug_bundle_file_path)) {
+        vlog(
+          lg.debug,
+          "Debug bundle file {} does not exist, cannot reload metadata",
+          debug_bundle_file_path);
+        co_await cleanup_files();
+        co_return co_await remove_kvstore_entry();
+    }
+
+    if (
+      run_was_successful
+      && !co_await validate_sha256_checksum(
+        debug_bundle_file_path, md.sha256_checksum)) {
+        vlog(
+          lg.debug,
+          "Debug bundle file {} checksum mismatch",
+          debug_bundle_file_path);
+        co_await cleanup_files();
+        co_return co_await remove_kvstore_entry();
+    }
+
+    vlog(
+      lg.info,
+      "Detected a valid previous run that was {}successful",
+      run_was_successful ? "" : "un");
+
+    std::optional<process_output> po;
+    auto process_output_file_exists = co_await ss::file_exists(
+      process_output_file_path);
+    if (process_output_file_exists) {
+        try {
+            auto buf = co_await read_file(process_output_file_path);
+            iobuf_parser p(std::move(buf));
+            po.emplace(serde::read<process_output>(p));
+        } catch (const std::exception& e) {
+            vlog(
+              lg.warn,
+              "Failed to read process output from {}: {}",
+              process_output_file_path,
+              e.what());
+        }
+    }
+
+    if (!po) {
+        vlog(
+          lg.warn,
+          "Failed to reload process output for job {} from {}.  Incomplete "
+          "metadata reload",
+          md.job_id,
+          process_output_file_path);
+        if (process_output_file_exists) {
+            co_await ss::remove_file(process_output_file_path);
```

Comment:
Why not `debug_bundle_file_path` or `kvstore_entry` cleanup here?

Reply:
Because in this situation, we are only missing the process stdout/stderr output.  So we are losing some information but the debug bundle is still valid.